pelican/rstdirectives.py

```diff
@@ -11,6 +11,16 @@
 import pelican.settings as pys
 
 
+def integers():
+    """Infinite sequence of integers."""
+    i = 1
+    while True:
+        yield str(i)
+        i = i + 1
+
+code_block_id = integers()
```

Comment:
You could also use `itertools.count(start=1)` to create a generator that yields the sequence of natural numbers. Also, would this still work if it was a class variable belonging to `Pygments` rather than a global?